### ISSUE TITLE
variable product with multiple variations issue

### ISSUE THREAD

**Issue by GeorgeWebDevCy** [GeorgeWebDevCy/gn-product-and-image-remover#1]
Hi there,
I was using this snippet, but I think there may be an issue with it.
Can you please test the following and see if you get the same results?

Create and publish a variable product with multiple variations, each with their own product image.
Use the 'Copy to a new draft' to create a copy of the variable product.

During the 'Copy to a new draft' process this snippet deletes all but the main product image, leaving all variations from the original product with no images.

I use the 'Copy to a new draft' a lot as most of my products are very similar, so making a copy and doing minor changes is the fastest way to create new products.
Unfortunately this issue makes the snippet completely unuseable, I'm just glad I spotted it relatively early and only had to reupload a few variable product images.

If you're able to replicate the issue, do you think it would easy to fix?

Thank you for your help!

**Comment by JapeNZ** [comment]
Following

**Comment by GeorgeWebDevCy** [comment]
@JapeNZ I assume that you are the person that reported the bug on Codesnippets Cloud. If I am correct could you test the plugin instead on the snippet? the plugin doesn't cause the issue for me. I am on the latest WordPress and all my plugins are up to date etc. Remember to disable the snippet when running the plugin [https://wordpress.org/plugins/gn-product-and-image-remover/](url)

Just wondering maybe it occurs only with the snippet and not the plugin

**Comment by JapeNZ** [comment]
Hi @GeorgeWebDevCy,
Thank you for taking a look at this for me :)

I've done some additional testing and the plugin works perfectly, infact I think the snippet might do as well.

I think what I was doing was creating a copy using the 'Copy to a new draft', and then using the 'Delete all variants' bulk action in the 'Variations' tab.
This deletes the variations so I can create new variations for the new product... but of course at the same time with your plugin (or snippet) active, it also deletes the images attached to the variations.

When I then check the original product that I had created a copy of, the variation images are missing.

Basically I need to remember to remove the images from the product variations in the product copy, before deleting those product variations and everything should work fine.

As far as I can tell, your plugin (and snippet) work perfectly... the error was me not using it correctly.

Thanks again for taking a look, I very much appreciate it :)

**Comment by GeorgeWebDevCy** [comment]
@JapeNZ sorry it took me so long to get back to you. My workload has been crazy.  I appreciate your feedback. I tried reproducing the bug but couldn't. Happy everything is working. Don't hesitate to drop a snippet/plugin idea my way. I like to tinker

**Comment by GeorgeWebDevCy** [comment]
@JapeNZ sorry it took me so long to get back to you. My work-load has been crazy.  I appreciate your feedback
